### PR TITLE
Add retries to the placement operations

### PR DESCRIPTION
**Pull request by halspang**
# Description

Add retries to the placement operations

## Issue reference

Please reference the issue this PR will close: #1914 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/placement/membership.go [hunk 1]
```diff
@@ -310,19 +312,32 @@ func (p *Service) disseminateOperation(targets []placementGRPCStream, operation
...
 	var err error
 	for _, s := range targets {
-		err = s.Send(o)
-
-		if err != nil {
-			remoteAddr := "n/a"
-			if peer, ok := peer.FromContext(s.Context()); ok {
-				remoteAddr = peer.Addr.String()
-			}
+		// TODO: Retry
```
Comment (artursouza) [inline pkg/placement/membership.go:315]: Why the TODO?

File: pkg/placement/membership.go [hunk 2]
```diff
@@ -310,19 +312,32 @@ func (p *Service) disseminateOperation(targets []placementGRPCStream, operation
...
+						remoteAddr = peer.Addr.String()
+					}
 
-			log.Errorf("error updating runtime host (%q) on %q operation: %s", remoteAddr, operation, err)
-			// TODO: the error should not be ignored. By handing error or retrying dissemination,
-			// this logic needs to be improved. Otherwise, the runtimes throwing the exception
-			// will have the inconsistent hashing tables.
-		}
+					log.Errorf("error updating runtime host (%q) on %q operation: %s", remoteAddr, operation, err)
+					// TODO: the error should not be ignored. By handing error or retrying dissemination,
```
Comment (artursouza) [inline pkg/placement/membership.go:331]: Do these comments still apply?

File: pkg/placement/membership.go (hunk 1 above)
Comment (halspang) [reply pkg/placement/membership.go:315]: Forgot to do a comment sweep, that was just a placeholder.

File: pkg/placement/membership.go (hunk 2 above)
Comment (halspang) [reply pkg/placement/membership.go:331]: Nope! Forgot to do a comment sweep.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**